### DISCUSSION TITLE
20231109 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#64] [Meetings]
Time: 8:30 UTC (10:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@WarmBeer) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by cgbosse** [comment]
# Links shared during the meeting

Mick
Mick says:
https://github.com/torrust/torrust-tracker/discussions/504
 
9:57
Jose Celano
Jose Celano says:
https://github.com/torrust/torrust-tracker/blob/develop/tests/servers/udp/client.rs
 
10:13
cam
cam says:
https://torrust.com/containerizing-rust-applications-best-practices
 
10:28
Jose Celano
Jose Celano says:
https://blog.rust-lang.org/
 
10:30
Mick
Mick says:
https://shoelace.style/
 
10:34
Jose Celano
Jose Celano says:
https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design
 
10:35
cam
cam says:
https://github.com/grail-ui/grail-ui
 
cam says:
https://grail-ui.vercel.app/svelte/clipboard
 
cam says:
https://github.com/rust-lang/rust/commit/fab1054e1742790c22ccc92a625736d658363677
 
cam says:
https://github.com/launchbadge/sqlx/issues/2855#issuecomment-1803567099
 
11:14
Mick
Mick says:
https://nl.steelcase.com/products/leap?gclid=Cj0KCQiAo7KqBhDhARIsAKhZ4ujldlCSvXq5sJSwvEbNKBFCxDFrTbjzhaQzodFlApWWrq1BmxJz5L8aAqciEALw_wcB&hsa_acc=6269880384&hsa_ad=&hsa_cam=18391046820&hsa_grp=&hsa_kw=&hsa_mt=&hsa_net=adwords&hsa_src=x&hsa_tgt=&hsa_ver=3&utm_campaign=%5BJP%5D%20-%20Sales-Performance%20Max%20%5BShopping%5D%20ecomm%20NL&utm_medium=ppc&utm_source=adwords&utm_term=&variant=41525302460584

**Comment by josecelano** [comment]
### Torrust Website

Finished article: [Containerizing Rust Applications](https://torrust.com/containerizing-rust-applications-best-practices)

**Comment by mario-nt** [comment]
### GUI

- Fix CI errors on https://github.com/torrust/torrust-index-gui/pull/306
- Worked on https://github.com/torrust/torrust-index-gui/pull/297
- Worked on https://github.com/torrust/torrust-index-gui/pull/298
- Research on deleting package-lock.json
- GUI not working after updating dependencies

### INDEX

- Worked on https://github.com/torrust/torrust-index/pull/367
- Worked on https://github.com/torrust/torrust-index/issues/39
- Worked on https://github.com/torrust/torrust-index/pull/367
- Work on https://github.com/torrust/torrust-index/pull/370
- Open new issue https://github.com/torrust/torrust-index/issues/371

### API Lib

- Worked on https://github.com/torrust/torrust-index-api-lib/pull/39

### Other

- Worked on https://github.com/torrust/torrust-website/issues/99

### Emails and discussions:

- Read https://github.com/torrust/torrust-index/issues/361
- Read https://github.com/torrust/torrust-index/issues/362
- Read [https://github.com/torrust/torrust-tracker/pull/473](https://github.com/torrust/torrust-tracker/pull/473#issuecomment-1785651750)
- Read https://github.com/torrust/torrust-tracker/issues/492
- Read https://github.com/torrust/torrust-website/discussions/96#discussioncomment-7480426
- GPG Key ready for verified commits

### Training/Research

- Rust videos https://www.youtube.com/@letsgetrusty 
- Cypress delete testing data after testing best practices
- Rust Read-Write Locks

**Comment by grmbyrn** [comment]
## Torrust Website
* Added profile pages and linked them to articles.
* Added a button to each code block so that it would copy that code.
* Researched and found people to talk to about how to make the button copy the right code.

## Python
* Continued learning Python on Codecademy.

## Svelte/SvelteKit
* Revised Svelte and SvelteKit.
* Began creating a website to host blogs on Svelte, SvelteKit, and other technologies I learn.

**Comment by mickvandijke** [comment]
### Torrust Tracker
- Looked into Rust profiling tools (RustRover/perf/flamegraph) and relevance for the tracker.
- Looked into benchmarking frameworks for Rust besides criterion.
- Tried several solutions of testing lock contention (in parallel code) using Criterion.
